jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/UrlEncoded.java

```diff
@@ -873,7 +872,14 @@ else if (buffer != null)
                 return encoded.substring(offset, offset + length);
             }
 
-            return buffer.toString();
+            try
+            {
+                return buffer.takeString();
+            }
+            catch (CharacterCodingException e)
+            {
+                throw new RuntimeException(e);
+            }
```

Comment:
```suggestion
            return buffer.takeString(() -> new RuntimeException("Bad UTF-8"));
```

Reply:
Currently the API taking a `Supplier<X extends Throwable>` is only on the `Utf8StringBuilder` interface and not on  `CharsetStringBuilder`.  We don't have it there because we have some implementations (i.e. the JVM Decoder ones) that are hard wired to throw `CharacterCodingException`, so we cannot implement an arbitrary supplier.